File: composeApp/src/androidMain/kotlin/com/kakapo/oakane/presentation/ui/model/CategoryIconMapper.kt
package com.kakapo.oakane.presentation.ui.model

import com.kakapo.oakane.R
import com.kakapo.oakane.model.category.CategoryIconName

fun CategoryIconName.asIcon(): Int {
    return when (this) {
        // INCOME
        CategoryIconName.SALARY -> R.drawable.ic_rounded_work
        CategoryIconName.FREELANCE -> R.drawable.ic_rounded_home_work
        CategoryIconName.BUSINESS_PROFITS -> R.drawable.ic_rounded_trending_up
        CategoryIconName.RENTAL_INCOME -> R.drawable.ic_outline_insights
        CategoryIconName.INTEREST -> R.drawable.ic_round_attach_money
        CategoryIconName.DIVIDENDS -> R.drawable.ic_rounded_show_chart
        CategoryIconName.GIFTS -> R.drawable.ic_outline_card_giftcard

        // HOUSING
        CategoryIconName.RENT_MORTGAGE -> R.drawable.ic_rounded_real_estate_agent
        CategoryIconName.UTILITIES -> R.drawable.ic_outline_lightbulb
        CategoryIconName.HOME_INSURANCE -> R.drawable.ic_outline_assignment
        CategoryIconName.PROPERTY_TAXES -> R.drawable.ic_round_attach_money
        CategoryIconName.HOME_MAINTENANCE -> R.drawable.ic_outline_maintenance
        CategoryIconName.HOA_FEES -> R.drawable.ic_outline_home

        // TRANSPORTATION
        CategoryIconName.CAR_PAYMENT -> R.drawable.ic_outline_payments
        CategoryIconName.GAS -> R.drawable.ic_outline_gas
        CategoryIconName.INSURANCE -> R.drawable.ic_outline_transportation_insurance
        CategoryIconName.PARKING_FEES -> R.drawable.ic_outline_local_parking
        CategoryIconName.PUBLIC_TRANSPORTATION -> R.drawable.ic_outline_directions_bus
        CategoryIconName.MAINTENANCE -> R.drawable.ic_outline_handyman

        // FOOD
        CategoryIconName.GROCERIES -> R.drawable.ic_outline_local_grocery_store
        CategoryIconName.RESTAURANTS -> R.drawable.ic_outline_restaurant_menu
        CategoryIconName.DINING_OUT -> R.drawable.ic_outline_fastfood

        // ENTERTAINMENT
        CategoryIconName.MOVIES -> R.drawable.ic_outline_movie
        CategoryIconName.CONCERTS -> R.drawable.ic_rounded_event
        CategoryIconName.SUBSCRIPTIONS -> R.drawable.ic_baseline_subscriptions
        CategoryIconName.HOBBIES -> R.drawable.ic_outline_emoji_events
        CategoryIconName.TRAVEL -> R.drawable.ic_outline_airplanemode_active

        // EDUCATION
        CategoryIconName.TUITION -> R.drawable.ic_outline_school
        CategoryIconName.BOOKS -> R.drawable.ic_outline_book
        CategoryIconName.SUPPLIES -> R.drawable.ic_outline_inventory_2

        // HEALTHCARE
        CategoryIconName.DOCTORS_VISITS -> R.drawable.ic_rounded_calendar_today
        CategoryIconName.PRESCRIPTIONS -> R.drawable.ic_rounded_local_pharmacy
        CategoryIconName.INSURANCE_PREMIUMS -> R.drawable.ic_outline_monetization_on
        CategoryIconName.DENTAL_CARE -> R.drawable.ic_rounded_health_and_safety
        CategoryIconName.VISION_CARE -> R.drawable.ic_outline_remove_red_eye

        // PERSONAL CARE
        CategoryIconName.HAIR -> R.drawable.ic_outline_face_retouching
        CategoryIconName.NAILS -> R.drawable.ic_rounded_spa
        CategoryIconName.CLOTHING -> R.drawable.ic_round_checkroom
        CategoryIconName.TOILETRIES -> R.drawable.ic_rounded_soap

        // SHOPPING
        CategoryIconName.GIFTS_EXPENSE -> R.drawable.ic_rounded_celebration
        CategoryIconName.CLOTHES -> R.drawable.ic_round_checkroom
        CategoryIconName.ELECTRONICS -> R.drawable.ic_rounded_devices
        CategoryIconName.HOUSEHOLD_ITEMS -> R.drawable.ic_rounded_weekend

        // DEBT
        CategoryIconName.CREDIT_CARD_PAYMENTS -> R.drawable.ic_rounded_credit_card
        CategoryIconName.LOAN_PAYMENTS -> R.drawable.ic_outline_payments
        CategoryIconName.INTEREST_EXPENSE -> R.drawable.ic_round_attach_money

        //Default
        CategoryIconName.DEFAULT -> R.drawable.ic_outline_account_balance_wallet
    }
}

File: iosApp/iosApp/designSystem/model/CategoryIconMapper.swift
import SwiftUI
import Shared

extension CategoryIconName {
    func asIconCategory() -> String {
        switch self {
        case .salary:
            return IconConstant.BriefcaseFill
        case .freelance:
            return IconConstant.PersonCropCircleBadgeCheckmark
        case .businessProfits:
            return IconConstant.ChartBarXAxis
        case .rentalIncome:
            return IconConstant.Building2CropCircle
        case .interest:
            return IconConstant.Percent
        case .dividends:
            return IconConstant.ChartLineUptrendXYAxis
        case .gifts:
            return IconConstant.GiftFill
        case .wallet:
            return IconConstant.WalletBifold
            
        case .rentMortgage:
            return IconConstant.BuildingColumnsFill
        case .utilities:
            return IconConstant.BoltFill
        case .homeInsurance:
            return IconConstant.ShieldFill
        case .propertyTaxes:
            return IconConstant.BanknoteFill
        case .homeMaintenance:
            return IconConstant.HammerFill
        case .hoaFees:
            return IconConstant.BanknoteFill
            
        case .carPayment:
            return IconConstant.CarFill
        case .gas:
            return IconConstant.FuelpumpFill
        case .insurance:
            return IconConstant.ShieldFill
        case .parkingFees:
            return IconConstant.ParkingFill
        case .publicTransportation:
            return IconConstant.BusFill
        case .maintenance:
            return IconConstant.WrenchFill
            
        case .groceries:
            return IconConstant.CartFill
        case .restaurants:
            return IconConstant.ForkKnife
        case .diningOut:
            return IconConstant.ForkKnifeCircle
            
        case .movies:
            return IconConstant.FilmFill
        case .concerts:
            return IconConstant.SpeakerFill
        case .subscriptions:
            return IconConstant.Calendar
        case .hobbies:
            return IconConstant.GamecontrollerFill
        case .travel:
            return IconConstant.MapFill
            
        case .tuition:
            return IconConstant.GraduationCapFill
        case .books:
            return IconConstant.BookFill
        case .supplies:
            return IconConstant.TrayFullFill
            
        case .doctorsVisits:
            return IconConstant.Stethoscope
        case .prescriptions:
            return IconConstant.PillsFill
        case .insurancePremiums:
            return IconConstant.ShieldCheckerboard
        case .dentalCare:
            return IconConstant.FaceSmilingFill
        case .visionCare:
            return IconConstant.Eyeglasses
            
        case .hair:
            return IconConstant.CombFill
        case .nails:
            return IconConstant.HandThumbsupFill
        case .clothing:
            return IconConstant.TShirt
        case .toiletries:
            return IconConstant.Shower
            
        case .giftsExpense:
            return IconConstant.Gift
        case .clothes:
            return IconConstant.BagFill
        case .electronics:
            return IconConstant.DesktopComputer
        case .householdItems:
            return IconConstant.HouseFill
            
        case .creditCardPayments:
            return IconConstant.CreditCard
        case .loanPayments:
            return IconConstant.BanknoteFill
        case .interestExpense:
            return IconConstant.ArrowUpRightCircle
        case .default:
            return IconConstant.AppleLogo
        }
    }
}
